### PR TITLE
feat: Add Cloudflare Access middleware

### PR DESCRIPTION
**Pull request by G4brym**
usage:

```
app.use('*', cloudflareAccess('my-access-team-name'))
```

access jwt payload
```
app.get('/', (c) => {
  const payload = c.get('accessPayload')
  
  return c.text(`You just authenticated with the email ${payload.email}`)
})
```

### CODE REVIEW

**Review by yusukebe — COMMENTED** [review]
Can you add a CI for this middleware like this: https://github.com/honojs/middleware/blob/main/.github/workflows/ci-hello.yml

And I've left comments.

**Comment by yusukebe** [comment]
Hi @G4brym 

I've tried this middleware in my environment, and it works well! Please check some comments.

File: packages/cloudflare-access/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+function getJwt(c: Context) {
+  const authHeader = c.req.header('cf-access-jwt-assertion')
+  if (!authHeader) {
+    return null
+  }
+  return authHeader.trim()
+}
+
+function decodeJwt(token: string): DecodedToken {
```
[the remark this answers is not in the document]
Comment (G4brym) [reply packages/cloudflare-access/src/index.ts:125]: Looking at the [decode() helper code](https://github.com/honojs/hono/blob/main/src/utils/jwt/jwt.ts#L102), it doesn't return the signature part of the jwt, i need that to be able to verify the signature in line 147

**Comment by G4brym** [comment]
> Can you add a CI for this middleware like this: https://github.com/honojs/middleware/blob/main/.github/workflows/ci-hello.yml
> 
> And I've left comments.

i've just added the CI

File: packages/cloudflare-access/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yusukebe) [reply packages/cloudflare-access/src/index.ts:125]: Okay!

**Comment by yusukebe** [comment]
@G4brym 

Using `CODEOWNER` sounds good! (Currently, we don't use it for this repo)

Looks good for this PR. Let's go. Thank you very much!

**Pull request merged**